### PR TITLE
feat: add plugins

### PR DESCRIPTION
**Pull request by RodEsp**
## What does this PR do?
Adds the plugins plugin to the hello-world example so that CLIs created with `oclif generate` will include it out-of-the-box.

Also updates some names to fit better for an executable instead of a library.

### CODE REVIEW

**Review by peternhale — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -32,7 +34,7 @@
   },
   "files": [
     "/bin",
-    "/lib",
+    "/dist",
```
Comment (mdonnalley) [inline package.json:37]: why this change?

File: package.json [hunk 2]
```diff
@@ -41,12 +43,15 @@
...
-  "main": "lib/index.js",
+  "main": "dist/index.js",
+  "dirname": "oex",
   "oclif": {
-    "commands": "./lib/commands",
     "bin": "oex",
+    "commands": "./dist/commands",
     "plugins": [
-      "@oclif/plugin-help"
+      "@oclif/plugin-autocomplete",
```
Comment (mdonnalley) [inline package.json:52]: Do we want to be opinionated about the plugins that people use? Or maybe it would suffice to add a list of useful plugins to the readme?

I ask because I think developers generally prefer to add functionality they want as opposed to removing functionality they don't want

File: package.json (hunk 1 above)
Comment (RodEsp) [reply package.json:37]: Just to make it more semantic about what is being built. This repo doesn't represent a library.

File: package.json (hunk 2 above)
Comment (RodEsp) [reply package.json:52]: Yeah, you're right. I'll remove this.

Comment (peternhale) [reply package.json:52]: Maybe `@oclif/plugin-commands` might be a useful addition to the hello world`

Comment (RodEsp) [reply package.json]: I agree with Mike, we should just make developers aware of those plugins but not include them in here so they don't have to remove them if they don't want them. Even if we think they're terribly useful.

File: package.json (hunk 1 above)
Comment (mdonnalley) [reply package.json:37]: @RodEsp Are you concerned that this is a divergence from every other oclif plugin and library? Or are you thinking we'd update all of those as well?

Comment (mdonnalley) [reply package.json:37]: I'm not opposed to the change - I just think that following a single less-than-ideal pattern is better than following multiple patterns.

Do you mind logging a WI for updating all the other repos so that we don't loose track of it?

**Review by mdonnalley — APPROVED** [review]
